### DISCUSSION TITLE
BINKA/RADA audio format are now supported

### DISCUSSION THREAD

**Discussion by 4sval** [4sval/FModel#422] [Announcement]
https://github.com/vgmstream/vgmstream/issues/1261

**Comment by GhostScissors** [comment]
You can now listen and save both .BINKA and .RADA audio format files as long as you have the **BINKA or RADA decoder in your .data folder** corresponding to your sound audio format [binkadec.zip](https://github.com/4sval/FModel/files/14970514/binkadec.zip) [radadec.exe](https://github.com/GhostScissors/RadA-Decoder/releases/download/1.0/radadec.exe)

**Reply by NIKITI4-ZET** [reply]
what script is needed if I want to massively export binka files via Fmodel and then convert binkadec.exe to WAV via a BAT file, if I just drag binka audio onto it, then nothing happens

**Reply by 4sval** [reply]
https://github.com/4sval/FModel/blob/dev/FModel/ViewModels/AudioPlayerViewModel.cs#L626

**Reply by LLD-Jo** [reply]
this isn't working on windows 10 64-bit, i get a different OS error
![image](https://github.com/user-attachments/assets/32aceedb-59a6-4e48-91ef-a47e690c825c)

**Reply by Kein** [reply]
> You can now listen and save both .BINKA and .RADA audio format files as long as you have the **BINKA or RADA decoder in your .data folder** corresponding to your sound audio format [binkadec.zip](https://github.com/4sval/FModel/files/14970514/binkadec.zip) [radadec.exe](https://github.com/GhostScissors/RadA-Decoder/releases/download/1.0/radadec.exe)

https://www.virustotal.com/gui/file/4d0022c501d61ea97495edd58645c82aa78461b08e6be2ae79fc37a7b7b3a7f2

**Reply by Dahus** [reply]
>binkadec.zip

And who is the author of this project?

**Reply by Krowe-moh** [reply]
ghost made that

**Reply by GhostScissors** [reply]
> ghost made that

Binka Decoder was **not** made by me but by a discord user "_happyme531_". I instead created the Rada Decoder which is **based** on the Binka Deoder

**Reply by Krowe-moh** [reply]
mb binka and rada confusion

**Comment by DMClues** [comment]
Hi - thanks very much for this - really helpful. I've come across this audio compression in Unreal Engine 5.2 and this binkadec won't convert to WAV (normal Binka encoded files can be converted no problem)

BINKA_1000_StreamCache_Ver5027_MEM_65536MaxChnkSize_00483224128

Have you seen this / any idea on whether it is convertable?

**Reply by 4sval** [reply]
this is just a regular binka file [with streaming and caching information](https://github.com/EpicGames/UnrealEngine/blob/5.2/Engine/Source/Runtime/AudioPlatformConfiguration/Public/AudioCompressionSettings.h#L103-L144) append to its format
also known as a [platform cook overrides](https://github.com/EpicGames/UnrealEngine/blob/072300df18a94f18077ca20a14224b5d99fee872/Engine/Source/Runtime/Engine/Private/SoundWave.cpp#L216-L217)

CUE4Parse already [handles](https://github.com/FabianFG/CUE4Parse/blob/master/CUE4Parse-Conversion/Sounds/SoundDecoder.cs#L34) these compression names, and so binkadec should work just fine
if it does not work, its output may give big hints as to why

**Reply by DMClues** [reply]
Thanks - not sure whether it was a reboot of PC, or upgrading to latest version of Fmodel 4.4.3.6, but these files are also now playing no problem :)

**Reply by Animeliqite** [reply]
Updating the program doesn't work for me btw. Tried putting the binkadec app into the .data directory but it still didn't work. For some reason, MonteCarlo isn't a .binka but a .bınka.

**Comment by ANDUNOTHISMAINN** [comment]
Hey, where can i find the "Rada" plug in for the audio in Fmodel, when i try to play something i get the error [WRN] Unsupported audio format 'RADA' please would someone kindly assist me please i cant locate the codecs/plug in.

**Comment by SpecterArdy** [comment]
https://www.virustotal.com/gui/file/c85f3bdf6a682e3bbc134931a62f92fed309b89781a68b8799183eb870360589

why is it showing up as Malware?
![image](https://github.com/user-attachments/assets/b7071ddf-7755-43af-ac19-16f26e405674)

**Reply by h4lfheart** [reply]
It's a false positive, this one should "fix" the false av pickup

https://github.com/h4lfheart/FortnitePorting/blob/v3/FortnitePorting/Assets/Dependencies/radadec.exe

**Comment by adastmin1** [comment]
![image](https://github.com/user-attachments/assets/63958f34-9eaf-4402-bb77-02023af3bee9)
new format binka?

**Reply by 595554963github** [reply]
https://github.com/vgmstream/vgmstream/issues/1826

**Comment by nekkk** [comment]


**Reply by Ka1serM** [reply]
> wav to binka

???

**Reply by GhostScissors** [reply]
Why would you want to be encode it back to BINKA and why do you want to add it FModel?

**Reply by nekkk** [reply]
> Why would you want to be encode it back to BINKA and why do you want to add it FModel?

Dubbing mod purposes.

**Reply by Ka1serM** [reply]
> > Why would you want to be encode it back to BINKA and why do you want to add it FModel?
> 
> Dubbing mod purposes.

Fmodel is a datamining only tool.

**Comment by Mermothle** [comment]
Just as a heads up to those who might wonder, the .binka -> wav conversion appears to fail if the filename has special characters in it, i.e. is chinese. I can get every audio file in that project working except the chinese-named files. If you're trying to listen to audio like that, you might be out of luck for now.

**Comment by RizGBlue** [comment]
Is there a way to convert files from .binka into .wav without playing them through FModel? Trying to extract all sounds from a game.

**Reply by wp760836** [reply]
> Is there a way to convert files from .binka into .wav without playing them through FModel? Trying to extract all sounds from a game.

yes there is. Just use binkadec.exe itself. If you want a batch process, just write a script to obtain it
